tendrl/commons/flows/base_flow.py

```diff
@@ -39,17 +38,10 @@ def __init__(
         self.uuid = uuid
         self.parameters = parameters
         self.job = job
-        self.config = config
         self.definitions = definitions
 
+        # Add etcd_client to parameters via child flows
         self.parameters.update({'log': []})
-        etcd_kwargs = {
-            'port': int(self.config.get("commons", "etcd_port")),
-            'host': self.config.get("commons", "etcd_connection")
-        }
-
-        self.etcd_client = etcd.Client(**etcd_kwargs)
-        self.parameters.update({'etcd_client': self.etcd_client})
```

Comment:
creating an etcd_client is costly (default creates a httpconnectionpool of 10), hence it is ideal that the rpc layer provides an instance of the etcd_client by appending it to parameters during flow invocation